api/src/test/java/gov/cms/ab2d/api/controller/AdminAPIIPTests.java

```diff
@@ -0,0 +1,214 @@
+package gov.cms.ab2d.api.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.okta.jwt.JwtVerificationException;
+import gov.cms.ab2d.api.SpringBootApp;
+import gov.cms.ab2d.common.dto.SponsorDTO;
+import gov.cms.ab2d.common.dto.SponsorIPDTO;
+import gov.cms.ab2d.common.model.Sponsor;
+import gov.cms.ab2d.common.repository.ContractRepository;
+import gov.cms.ab2d.common.repository.JobRepository;
+import gov.cms.ab2d.common.repository.RoleRepository;
+import gov.cms.ab2d.common.repository.SponsorRepository;
+import gov.cms.ab2d.common.repository.UserRepository;
+import gov.cms.ab2d.common.util.AB2DPostgresqlContainer;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.List;
+import java.util.Set;
+
+import static gov.cms.ab2d.common.util.Constants.ADMIN_PREFIX;
+import static gov.cms.ab2d.common.util.Constants.ADMIN_ROLE;
+import static gov.cms.ab2d.common.util.Constants.API_PREFIX;
+import static gov.cms.ab2d.common.util.Constants.SPONSOR_ROLE;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+@SpringBootTest(classes = SpringBootApp.class, webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@AutoConfigureMockMvc
+@Testcontainers
+public class AdminAPIIPTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Container
+    private static final PostgreSQLContainer postgreSQLContainer = new AB2DPostgresqlContainer();
+
+    @Autowired
+    private SponsorRepository sponsorRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Autowired
+    private JobRepository jobRepository;
+
+    @Autowired
+    private ContractRepository contractRepository;
+
+    @Autowired
+    private TestUtil testUtil;
+
+    private String token;
+
+    private static final String IP_URL = "/ip";
+
+    @BeforeEach
+    public void setup() throws JwtVerificationException {
+        contractRepository.deleteAll();
+        jobRepository.deleteAll();
+        userRepository.deleteAll();
+        roleRepository.deleteAll();
+        sponsorRepository.deleteAll();
```

Comment:
Move cleanup to an @AfterEach method, this improves the chances of not interfering with other tests.

Reply:
Or just duplicate in an aftereach method

Reply:
Probably duplicating is the best for now, moving to aftereach just means it may not get cleaned up for this particular suite. Eventually there should be something that just deletes all of the data in each suite in the Before but it's not consistent now between test suites.

Reply:
I don't get why one would ever want to delete things in the before.  Stale data should be fixed where it is created, not in all other places affected by it being left around.

I also don't get how @AfterEach would fail to cleanup in the suite.  One could add the other form of @After, which works on a class level to do the final cleanup.

Reply:
You could pick either one, but it seems easier to do it in the before since there's already setup involved there. As long as every test does it in a before then no test would be affected by data. If there's data remaining when this suite starts then it would effect the current suite.

Reply:
Putting it in the before (in another example) caused other, unrelated tests to fail:  see the solution to https://jira.cms.gov/browse/AB2D-1951.

Reply:
See the checkin here: https://github.com/CMSgov/ab2d/commit/2baca1f153a5a49895c4746401841a77302ce41e

Reply:
Cleaning up and deleting from tables shouldn't cause others to fail unless they're not setting up data properly.

Reply:
Whichever test runs first is somewhat random along with other tests, so it's hard to predict if something else would have cleaned up at this point. Until it's standardized you can't really predict which one will go first

Reply:
My point is that another class should never have leftover data.  Each class ought to clean up after itself.